### DISCUSSION TITLE
How to manage homebrew package of Zim (that works on M1s)

### DISCUSSION THREAD

**Discussion by reagle** [zim-desktop-wiki/zim-desktop-wiki#1749] [General]
@dehesselle is kindly providing an "official" [MacOS DMG file](https://zim-wiki.org/downloads.html) for zim, which works with M1 Macs. As discussed in #1712, he will not be providing a homebrew package. Questions:

1. What should be done with the [existing brew](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/zim.rb)? I assume it still works well for folks on x86, but it will frustrate and confuse M1 users. I suggest it be removed and we create a binary cask instead using the officially released DMG file.
2. Who "owns" the existing brew? Can we delete it in favor of a cask?
3. Does anyone have experience [creating a cask](https://nicedoc.io/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md)? It *looks* straightforward, and I've used `bump-cask-pr` to request version updates to existing casks, but am otherwise a newbie.

**Comment by dehesselle** [comment]
If creating a cask is rather simple, that is something I can look into, but it's not my top priority TBH. I guess once a cask is created (sorry can't look at it myself right now), keeping it updated is a matter of changing a link, a version number and a checksum - basically low maintenance.

(My concern only lies with questions/issues like "xyz doesn't work with my non-cask installation of Zim from Homebrew", which are beyond my scope/capacity.)

**Comment by reagle** [comment]
Yes, changing a cask is done via a simple command:

    brew bump-cask-pr --version <new_version> <outdated_cask>

I've done that before for packages/casks I have nothing to do with as it is largely automated.

**Comment by jaap-karssenberg** [comment]
Sorry, but I'm converting this to a discussions. Discussions belong in the forum, not in the issue tracker.

**Comment by reagle** [comment]
BTW: I just tried installing zim--0.74.3 via homebrew and no Zim.app appeared in my /Applications folder. (So I download the DMG file and manually installed it.)

**Comment by reagle** [comment]
@dehesselle, any suggestion on who might I ask about the odd behavior above?

**Reply by dehesselle** [reply]
You mean that the Homebrew install does not appear in `/Applications`? - Homebrew only installs the Python files from source and does not provide an application bundle. A cask is required for that (which in turn could point to our app) if I am not mistaken.

**Reply by reagle** [reply]
Yes, and therefore wasn't accessible to my launcher. I didn't realize brews can't create symlinks in /Applications.

**Comment by introt** [comment]
> Who "owns" the existing brew?

I don't think there's one "owner" - looking at [the bottle's commit history](https://github.com/Homebrew/homebrew-core/commits/06fc3a13d8cc517d6b33c766623cf17231902127?before=06fc3a13d8cc517d6b33c766623cf17231902127+35&branch=06fc3a13d8cc517d6b33c766623cf17231902127&path%5B%5D=Formula&path%5B%5D=zim.rb), there seem to be multiple maintainers.

I suggest you take this issue up at [Homebrew/discussions](https://github.com/orgs/Homebrew/discussions) - the people there seem quite helpful, and more importantly include the ones in charge of the brew.


> Does anyone have experience [creating a cask](https://nicedoc.io/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md)?

The quickest way would be to contact the Homebrew folk via https://github.com/orgs/Homebrew/discussions/categories/casks

**Comment by reagle** [comment]
@dehesselle, I'm wondering about the schedule of the Mac M1 builds? Are they in process for 0.75?

**Reply by dehesselle** [reply]
Yes, I'm working on it (finishing touches / one issue left). Release likely happens later today or tomorrow and I'll leave a comment on Jaap's release announcement discussion.
